src/hw_1/hw_1.ts

```diff
@@ -0,0 +1,124 @@
+class School {
+    directions: Direction[] = [];
+  
+    addDirection(direction: Direction): void {
+      this.directions.push(direction);
+    }
+  }
+  
+class Direction {
+    private _name: string;
+    levels: Level[] = [];
+  
+    get name(): string {
+      return this._name;
+    }
+  
+    constructor(name: string) {
+      this._name = name;
+    }
+  
+    addLevel(level: Level): void {
+      this.levels.push(level);
+    }
+  }
+  
+class Level {
+    private _name: string;
+    private _program: string;
+    groups: Group[] = [];
+  
+    constructor(name: string, program: string) {
+      this._name = name;
+      this._program = program;
+    }
+  
+    get name(): string {
+      return this._name;
+    }
+  
+    get program(): string {
+      return this._program;
+    }
+  
+    addGroup(group: Group): void {
+      this.groups.push(group);
+    }
+  }
+  
+class Group {
+    private _students: Student[] = [];
+    directionName: string;
+    levelName: string;
+  
+    get students(): Student[] {
+      return this._students;
+    }
+  
+    constructor(directionName: string, levelName: string) {
+      this.directionName = directionName;
+      this.levelName = levelName;
+    }
+  
+    addStudent(student: Student): void {
+      this._students.push(student);
+    }
+  
+    showPerformance(): Student[] {
+      const sortedStudents = [...this.students].sort(
+        (a, b) => b.getPerformanceRating() - a.getPerformanceRating()
```

Comment:
Kind reminder: It can be useful to add type annotations to function parameters and constants. Even if it's not strictly related to type security, it can help in navigating through the models used in callbacks.